Separate from-to ratios

rkfg opened the pull request:
This PR add equal treatment for `-f` flag allowing to specify the `from` channel by its index, but most importantly it adds separate `from` and `to` ratios. The use case is as follow:
- often the channel capacities are vastly different, you might have one channel at 95% and want to balance it down to 80% and you have some much bigger channel at 40%
- after this rebalance the bigger channel's balance would only become 45% so it's totally fine
- the current `-r` parameter allows to exclude slightly unbalanced channels and only consider those at their limit but it's applied to both source and target channels which is not always desired
- in the situation described above if we use `-r 20` the big channel with 45% outbound liquidity wouldn't even be considered even though it can safely accept those 15% from a smaller channel and still be balanced

Since there's probably no universal solution that fits everyone I propose to specify the desired inbound/outbound ratio explicitly. If you run rebalancing automatically you can use these new flags as follow:
- `rebalance.py -f 1 --ratio-from 20` — would choose the first channel with less than 20% inbound liquidity and pick any target channel with less than 50% outbound liquidity (that can accept the amount without becoming unbalanced of course)
- `rebalance.py -t 1 --ratio-to 20` — would choose the first channel with less than 20% outbound liquidity and pick any source channel with less than 50% inbound liquidity
- `rebalance.py -t 1 --ratio-to 20 --ratio-from 20` — would choose the first channel with less than 20% outbound liquidity and pick any source channel with less than 20% inbound liquidity

Since by default the ratio is 50%, you only need to set the ratio for the direction you're interested in. But in some cases you might also limit the target channel set to shift the balance to the starving channels first. It all depends on the use case. The `-r` behavior is unchanged, it sets both `--ratio-to` and `--ratio-from` to the specified value.

PS: I also added `public_only=True` to the channel filter so that private channels are not used at all. By definition they can't (and should not) be balanced as they're client's channels.

accumulator commented:
I agree, this is a very nice improvement. 

And thanks for again a high-quality patch!

rkfg commented:
I changed some typos in the PR description so it's less confusing, I also rebased it on the current master (before you merged it so everything's fine). Thank you for improving the original script in the first place, this version provides a much better insight (and is prettier too!)
